tools/@aws-cdk/prlint/lint.ts

```diff
@@ -16,28 +13,81 @@ class LinterError extends Error {
   }
 }
 
-function createGitHubClient() {
-  const token = process.env.GITHUB_TOKEN;
+export interface PRLinterProps {
+  readonly client: any;
+  readonly owner: string;
+  readonly repo: string;
+  readonly number: number;
+}
+
+export class PRLinter {
+  private readonly client: any;
+  private readonly prParams: { owner: string, repo: string, pull_number: number };
+
 
-  if (token) {
-    console.log("Creating authenticated GitHub Client");
-  } else {
-    console.log("Creating un-authenticated GitHub Client");
+  constructor(private readonly props: PRLinterProps) {
+    this.client = props.client;
+    this.prParams = { owner: props.owner, repo: props.repo, pull_number: props.number };
   }
 
-  return new GitHub({'token': token});
+  public async validate() {
+    const number = this.props.number;
+
+    console.log(`⌛  Fetching PR number ${number}`);
+    const pr = (await this.client.get(this.prParams)).data;
+
+    console.log(`⌛  Fetching files for PR number ${number}`);
+    const files = (await this.client.listFiles(this.prParams)).data;
+
+    console.log("⌛  Validating...");
+
+    try {
```

Comment:
I think currently we exit as soon as one of the checks fail. I think it would be nice if we ran through all the checks and reported  _all_ of the failures at once.

Reply:
I also wonder if we should just do a single comment that we keep updating (or delete/create) with the checks that were run. It might be useful to see _all_ the check results even if they are not run. Maybe seeing that a specific check didn't run might prompt the reviewer to think about it?

> PR Linter Results: 
>  - ❌ Must contain a change to an integration test file. 
>  - ✅ Must contain a change to a test file.
>  - ❕Must contain a change to a README file. (not checked)
> PRs must pass status checks before we can provide a meaningful review.

Reply:
I like this! I'll make the change.